yarn-project/foundation/src/serialize/types.ts

```diff
@@ -0,0 +1,61 @@
+/**
+ * Represents a fixed-length array.
+ */
+export type TupleOf<T, N extends number> = N extends N ? (number extends N ? T[] : _TupleOf<T, N, []>) : never;
```

Comment:
What do you think about using this version instead, which seems simpler?

https://stackoverflow.com/a/71700658

Reply:
Tried it, broke everything related to normal arrays, not sure why but reverting